src/util/articleRenderers.ts

```diff
@@ -107,6 +106,55 @@ export const renderDesktopArticle = (json: any, articleId: string, articleDetail
     return json // This is HTML probably (the problem is that this is hard to know at this stage, buggy architecture)
 };
 
+export const getSectionsHtml = (sections: any, dump: Dump): string => {
+    let html = '';
+    let prev = 0;
+    let tocLevel = sections[0].toclevel;
+    let sectionsToSplit: any[] = [];
+
+    // Store index of sections on current level.
+    sections.map((section: any, index: number) => {
+        if (tocLevel >= section.toclevel) {
+            tocLevel = section.toclevel;
+            sectionsToSplit.push(index);
+        }
+    });
+
+    // Add first section on current level directly and remove it from list.
+    html += sectionTemplate({
+        section_index: sections[0].toclevel,
+        section_id: sections[0].id,
+        section_anchor: sections[0].anchor,
+        section_line: sections[0].line,
+        section_text: sections[0].text,
+        strings: dump.strings,
+    });
+
+    sectionsToSplit = sectionsToSplit.slice(1);
+
+    // Loop over sections on current level and recursively generate html for it's sub-section.
+    sectionsToSplit.forEach((currentIndex: number) => {
```

Comment:
I think you can just replace all of this by a loop which does not go through all items one by one, but goes through all items via list of slices of items. Each slice gathering the first item and the smaller ones which follow. Something like this:
```
foreach slice = getNextSectionSlice(sections) {
  sectionTemplate({
        section_index: slice[0].toclevel,
        section_id: slice[0].id,
        section_anchor: slice[0].anchor,
        section_line: slice[0].line,
        section_text: slice[0].text,
        strings: dump.strings,
        subsections_test: getSectionsHtml(slice[1..n])
});
}
```

I see no reason why you should have a specail treatment